src/Shared/Targets/Wrappers/AspNetBufferingTargetWrapper.cs

```diff
@@ -0,0 +1,305 @@
+        {
+            if (context == null)
+            {
+                return;
+            }
+
+            var bufferDictionary = GetBufferDictionary(context);
+
+            // If the dictionary is missing, create that first
+            if (bufferDictionary == null)
+            {
+                SetBufferDictionary(context);
+            }
+        }
+
+        /// <summary>
+        /// Called from the HttpModule or Middleware upon the end of the HTTP pipeline
+        /// Flushes all instances of this class registered in the HttpContext
+        /// </summary>
+        /// <param name="context"></param>
+        internal static void OnEndRequest(HttpContextBase context)
+        {
+            var bufferDictionary = GetBufferDictionary(context);
+            if (bufferDictionary == null)
+            {
+                return;
+            }
+            foreach (var bufferKeyValuePair in bufferDictionary)
+            {
+                bufferKeyValuePair.Key?.Flush(bufferKeyValuePair.Value);
+            }
+        }
+
+        /// <summary>
+        /// Called by the above static Flush method.
+        /// </summary>
+        /// <param name="buffer"></param>
+        private void Flush(Internal.LogEventInfoBuffer buffer)
+        {
+            if (buffer == null)
+            {
+                return;
+            }
+            InternalLogger.Trace("Sending buffered events to wrapped target: {0}.", WrappedTarget);
+            WrappedTarget?.WriteAsyncLogEvents(buffer.GetEventsAndClear());
+        }
+
+        /// <summary>
+        /// Obtains a slot in the buffer dictionary for 'this' class instance
+        /// If that does not exist, that is created first
+        /// if the dictionary does not exist, that is created first
+        /// </summary>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        private Internal.LogEventInfoBuffer GetOrCreateRequestBuffer(HttpContextBase context) 
+        {
+            // If the context is missing, stop
+            if (context == null)
+            {
+                return null;
+            }
+
+            var bufferDictionary = GetBufferDictionary(context);
+
+            // If the dictionary is missing, stop
+            if (bufferDictionary == null)
+            {
+                return null;
+            }
+
+            lock (bufferDictionary)
+            {
+                if (!bufferDictionary.TryGetValue(this, out var buffer))
+                {
+                    buffer = new Internal.LogEventInfoBuffer(BufferSize, GrowBufferAsNeeded, BufferGrowLimit);
+                    bufferDictionary.Add(this, buffer);
+                }
+
+                return buffer;
+            }
+        }
+
+        /// <summary>
+        /// The Key for the buffer dictionary in the HttpContext.Items collection
+        /// </summary>
+        private static readonly object HttpContextItemsKey = new object();
```

Comment:
Can this be renamed to `private readonly object dataSlot = new object();` and be moved to the top of the class (without XML-docs)

Again to reduce the changes to make the rename automatic.